### ISSUE TITLE
Getter functions unsupported as members of an object

### ISSUE THREAD

**Issue by esb** [ruby2js/ruby2js#216]
There doesn't seem to be a way to generate a getter function as a member of an object.

For example -

```ruby
obj = { getter:  -> () { return true } }
```

This generates

```js
obj = { getter() { return true }};
```

It doesn't seem possible to generate

```js
obj = { get getter() { return true }};
```

The intent is to be able to write

```js
let a = obj.getter;
```

**Comment by esb** [comment]
It looks like this is not being parsed correctly as well if there are multiple getter functions -

```ruby
 qi = {
          id: item.id,
          isSingleUse: -> () { item.isSingleUse },
          haveAnyCharges: -> () { self.item.isCharged && Number.isFinite(self.maxCharge) },
          maxCharge: -> () { item.maxCharges },
          charges: -> () { self.item.charges }
        }
```
What we get as output is
```js
let qi = {
  id: item.id,
  isSingleUse() { return item.isSingleUse },
  haveAnyCharges: () => this.item.isCharged && Number.isFinite(this.maxCharge),
  maxCharge() { return item.maxCharges },
  charges: () => this.item.charges
};
```

It appears that every second lambda is not being transformed.

**Comment by esb** [comment]
I guess the best answer is to convert the JS object into a Ruby class and define the getters there, rather than trying to find a way to specify JS getters/setters within the Ruby hash syntax, which really doesn't work.

**Comment by jaredcwhite** [comment]
@esb Yeah, that's a weird one…I would say it's unlikely I can investigate further any time soon so I'd just define regular class.

**Comment by esb** [comment]
The difference in the generated output is determined by the presence of ```self``` (this) in the function.

If ```self```` is present, then the output is of the form

```js
{
   name: () => ....
}
```

**Comment by esb** [comment]
As a workaround, the following works, albeit a bit clumsily.

Creating a getter in an object requires a full function definition if the function requires access to ```this``` in order to access values within the object. An arrow function does not provide a ```this```.

This means that the function needs to be defined first using the full function declaration syntax.
```ruby
def available()
  return self.current - self.used
end
```
The object can be defined as normal, but the getter value is not added.
```ruby
obj = {
  current: 10,
  used: 5
}
```
Finally, a getter is defined on the object.
```ruby
 Object.defineProperty(obj, "available", get: available)
```
We can then execute
```javascript
console.log(obj.available) => 5
```

**Comment by rubys** [comment]
## Proposed Solution

A clean way to support this would be to add a transformation to the **functions filter** that converts `Class.new { ... }.new` (instantiation of an anonymous class) into an object literal with getters/setters.

### Ruby syntax

```ruby
obj = Class.new do
  def value
    @value
  end
  
  def value=(v)
    @value = v
  end
end.new
```

### Current output

```javascript
let obj = new class {
  get value() { return this._value };
  set value(v) { this._value = v }
}
```

### Proposed output

```javascript
let obj = {
  get value() { return this._value },
  set value(v) { this._value = v }
}
```

### Rationale

1. **`new class { }` is rarely idiomatic JavaScript** - Creating an instance of an anonymous class is unusual since you can only ever create one instance. A plain object literal with getters/setters is more natural.

2. **Semantic equivalence for single instances** - For a one-off object, `{ get x() {...} }` and `new class { get x() {...} }` are functionally equivalent.

3. **Functions filter is the right place** - This filter already handles various method/function transformations, and this fits that pattern.

4. **Existing infrastructure** - The hash converter already supports outputting `get`/`set` syntax when given `:prop` nodes (see `hash.rb` lines 68-88). The filter would transform the AST to use this existing mechanism.

### Implementation approach

The functions filter would:
1. Detect the pattern: `:send` with method `:new`, receiver is a `:block` containing `Class.new`
2. Transform class method definitions into hash pairs with `:prop` keys
3. Output a hash node instead of the class instantiation

This gives users a clean Ruby idiom (`Class.new { }.new`) that maps to idiomatic JavaScript object literals with getters/setters.

**Comment by rubys** [comment]
Rich object literals is implemented and documented: https://www.ruby2js.com/docs/users-guide/patterns#rich-object-literals
